src/BookResourceStore.ts

```diff
@@ -0,0 +1,97 @@
+import Manifest from "Manifest";
+
+export interface IBookStore {
+  href: string;
+  data: Blob;
+}
+
+// IndexDB store for Book resources
+// Only works on browsers that support IndexedDB.
+export default class BookResourceStore {
+  private readonly db: IDBDatabase;
+
+  public constructor(db: IDBDatabase) {
+    this.db = db;
+  }
+
+  public static createBookResourceStore(): Promise<BookResourceStore> {
+    if (!("indexedDB" in window)) {
+      //TODO: a polyfill to handle lack of IndexedDB
+      console.log("This browser doesn't support IndexedDB");
+    }
+    var request = window.indexedDB.open("WebpubViewerDb", 2);
+    let store:BookResourceStore;
+    return new Promise((resolve) => {
+      request.onupgradeneeded = (evt) => {
+        let db = (<any>evt.target).result;
+        if (!db.objectStoreNames.contains("bookResources")) {
+          var bookResourceOS = db.createObjectStore("bookResources", {
+            keyPath: "href",
+          });
+          bookResourceOS.transaction.oncomplete = () => {
+            store = new BookResourceStore(db);
+          }
+        }
+      }
+
+      request.onsuccess = (evt) => {
+        if(!store) {
+          store = new BookResourceStore((<any>evt.target).result);
```

Comment:
Theoretically, yes.  Practically, I can't figure out what the type name for `IDBVersionChangeEvent.Target` is ... because `IDBVersionChangeEvent` doesn't have a target value documented.  Clearly it exists, but I can't find any mention of it in the documentation